src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -158,6 +163,19 @@
             Setting.Property.NodeScope,
             Setting.Property.Dynamic);
 
+    public static final Setting<Long> MODEL_CACHE_SIZE_IN_BYTES_SETTING = new Setting<>(MODEL_CACHE_SIZE_IN_BYTES,
```

Comment:
Why not use ```Setting.longSetting``` and provide default, min and max and avoid explicit validations just like above settings?

Reply:
Good question. `longSetting` didnt actually allow you to input a max limit in beta1 (until https://github.com/opensearch-project/OpenSearch/commit/b1b563dc39c39625b709d7f1ef97459f0e3b4e34 which is in rc1). I implemented the logic on my own. I will update